modules/biobank/php/containercontroller.class.inc

```diff
@@ -0,0 +1,405 @@
+ * LORIS Biobank Module.
+ *
+ * PHP Version 7.2
+ *
+ * @category   Controller
+ * @package    Loris
+ * @subpackage Biobank
+ * @author     Henri Rabalais <henri.rabalais@mcin.ca>
+ * @license    http://www.gnu.org/licenses/gpl-3.0.text GPLv3
+ * @link       http://www.github.com/aces/Loris/
+ */
+namespace LORIS\biobank;
+
+/**
+ * This class features the code that enables access to the containers of the
+ * LORIS Biobank Module.
+ *
+ * PHP Version 7.2
+ *
+ * @category   Controller
+ * @package    Loris
+ * @subpackage Biobank
+ * @author     Henri Rabalais <henri.rabalais@mcin.ca>
+ * @license    http://www.gnu.org/licenses/gpl-3.0.text GPLv3
+ * @link       http://www.github.com/aces/Loris/
+ */
+class ContainerController
+{
+    /**
+     * Global variable $db
+     *
+     * @access private
+     */
+    private $db;
+
+    /**
+     * Global variable $user
+     *
+     * @access private
+     */
+    private $user;
+
+    /**
+     * Global variable $dao
+     *
+     * @access private
+     */
+    private $dao;
+
+    /**
+     * Class Constructor
+     *
+     * @param Database $db   Database Instance.
+     * @param User     $user User Instance.
+     */
+    function __construct(\Database $db, \User $user)
+    {
+        $this->db   = $db;
+        $this->user = $user;
+        $this->dao  = $this->getDataProvisioner();
+    }
+
+    /**
+     * Get all Container Objects permitted by the Container Data Provisioner (DAO).
+     *
+     * @return array $specimens All permissable Container Objects
+     */
+    public function getInstances() : array
+    {
+        $this->validatePermission('GET');
+        $containers  = [];
+        $containerIt = $this->dao->execute($this->user);
+        foreach ($containerIt as $id => $container) {
+            $containers[$id] = $container;
+        }
+
+        return $containers;
+    }
+
+    /**
+     * This method takes in a list of container arrays from the front-end,
+     * validates them, instantiates them and then sends them to the DAO to be
+     * saved to the database.
+     *
+     * @param array $list A list of container arrays to be created.
+     *
+     * @return array $containerIds A list of IDs from the containers that were
+     *                             created.
+     */
+    public function createInstances(array $list) : array
+    {
+        $this->validatePermission('POST');
+        $containerList = array();
+        foreach ($list as $i => $item) {
+            $container = $this->dao->getInstanceFromArray($item);
+            $this->validateInstance($container);
+            $containerList[$i] = $container;
+        }
+
+        // Save Containers
+        $containerIds = [];
+        foreach ($containerList as $i => $container) {
+            $containerIds[$i] = $this->dao->saveInstance($container);
+        }
+
+        return $containerIds;
+    }
+
+    // TODO: Decide if this function should take a list of containers, rather
```

Comment:
You could always make a `updateInstances` that is a for-loop wrapper around this one.